libraries/lib-utility/Observer.h

```diff
@@ -0,0 +1,231 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file Observer.h
+  @brief Publisher-subscriber pattern, also known as Observer
+
+  Paul Licameli
+
+**********************************************************************/
+#ifndef __AUDACITY_OBSERVER__
+#define __AUDACITY_OBSERVER__
+
+#include <cassert>
+#include <functional>
+#include <memory>
+#include <type_traits>
+
+namespace Observer {
+
+// See below
+template<typename Message, bool NotifyAll>
+class Publisher;
+
+//! Default message type for Publisher
+/*!
+ This has a virtual destructor so that it can be the operand of dynamic_cast,
+ but that is not a requirement for Publisher's template argument.
+ */
+struct UTILITY_API Message {
+public:
+   ~Message();
```

Comment:
The comment above claims the destructor to be virtual

Reply:
will fix